### PR TITLE
Ruby: Account for `protected` methods in call graph

### PR DESCRIPTION
**Pull request by hvitved**
https://github.com/github/codeql/pull/10002 follow-up.

### CODE REVIEW

**Review by aibaars — COMMENTED** [review]
This looks good to me. Could you extend the test cases a bit?

File: ruby/ql/test/library-tests/modules/calls.rb [hunk 1]
```diff
@@ -447,7 +447,7 @@ def m5
 ConditionalInstanceMethods.new.m3 # currently unable to resolve
 ConditionalInstanceMethods.new.m4 # currently unable to resolve
 ConditionalInstanceMethods.new.m5 # NoMethodError
-exit
+
```
[the remark this answers is not in the document]
Comment (hvitved) [reply ruby/ql/test/library-tests/modules/calls.rb:450]: The `exit` call was never meant to be there in the first place, it was just a stray debugging line.

File: ruby/ql/test/library-tests/modules/calls.rb [hunk 2]
```diff
@@ -483,3 +483,17 @@ def singleton
...
+    protected def foo
+        puts "ProtectedMethods#foo"
+    end
+
+    def bar
+        foo
+        ProtectedMethods.new.foo
+    end
+end
+
```
[the remark this answers is not in the document]
Comment (hvitved) [reply ruby/ql/test/library-tests/modules/calls.rb:510]: Done.

File: ruby/ql/test/library-tests/modules/calls.rb [hunk 3]
```diff
@@ -484,16 +487,37 @@ def singleton
...
 end
 
 ProtectedMethods.new.foo # NoMethodError
-ProtectedMethods.new.bar
+ProtectedMethods.new.bar # NoMethodError
+ProtectedMethods.new.baz
+
+class ProtectedMethodsSub < ProtectedMethods
+    def baz
+        foo
```
Comment (aibaars) [inline ruby/ql/test/library-tests/modules/calls.rb:517]: Also add
```
ProtectedMethodsSub.new.foo
```

**Pull request merged**